javascript/tables/src/guessDataTypeByColumn.ts

```diff
@@ -0,0 +1,37 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import { DataType } from '@datashaper/schema'
+
+import { guessDataType } from './guessDataType.js'
+
+export function guessDataTypeByColumn(columnValues: string[]): DataType {
+	const guesser = guessDataType()
+
+	const mapTypes = new Map()
+
+	columnValues.forEach(value => {
+		const dataTypeResult = guesser(value)
+
+		if (dataTypeResult !== DataType.Null && !mapTypes.has(dataTypeResult)) {
```

Comment:
This seems overly complex. I would suggest iterating each value and simply putting the type in the map with `map.set`, including `DataType.Null` when the value is null/undefined/nan. At the end should be all the logic for determining the final type for the column: (1) delete DataType.Null, if there are no entries left the column type is null, (2) if there is more than one type remaining, the type is string, (3) otherwise return the remaining type